### PR TITLE
fix(macro): global object to globalThis

### PR DESCRIPTION
**Pull request by thewtex**
`global` is not necessarily defined in browsers:

  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis

Update in macro.js.

### CODE REVIEW

**Comment by thewtex** [comment]
@floryst @jourdain I encountered the `global not defined` issue in the more rare context of ES modules and iframes.

**Comment by finetjul** [comment]
Can't we think of a better variable name for `shimmedGlobalThis` ? `vtkGlobal` or maybe `macro.global` (eventually we will also have macro.window ?)

File: package-lock.json [hunk 1]
```diff
@@ -7524,7 +7524,6 @@
       "version": "1.1.3",
       "resolved": "https://registry.npmjs.org/define-properties/-/define-properties-1.1.3.tgz",
       "integrity": "sha512-3MqfYKj2lLzdMSf8ZIZE/V+Zuy+BgD6f164e8K2w7dgnpKArBDerGYpM46IYYcjnkdPNMjPk9A6VFB8+3SKlXQ==",
-      "dev": true,
```
Comment (finetjul) [inline package-lock.json:7527]: This is a genuine question: why get rid of "dev" ? It shouldn't have been here in the first place ?

**Comment by thewtex** [comment]
> Can't we think of a better variable name for shimmedGlobalThis ? vtkGlobal or maybe macro.global (eventually we will also have macro.window ?)

Sure, we can use `vtkGlobal`. Note that this variable is not exported, though.

File: package-lock.json (hunk 1 above)
Comment (thewtex) [reply package-lock.json:7527]: This file is auto-generated by `npm` and updates with their policies.

**Comment by finetjul** [comment]
It looks good to me but I don't know much about that "global" issue.

Yet, I wonder if we shouldn't wrap `globalThisShim()` into a `macro.js` function to make the dependency less sensitive to future changes.

**Comment by thewtex** [comment]
> Yet, I wonder if we shouldn't wrap globalThisShim() into a macro.js function to make the dependency less sensitive to future changes.

Yes, if we find that we need it in more contexts, we could make it available in `macro.js`. The examples use `global`, but this shim is not required there. In general, we want to avoid using `global` unless necessary as a best practice. So, we can avoid making it easy to do the wrong thing. At this point, it may be a cart before the hoarse. :shopping_cart: :hoarse:

**Comment by agirault** [comment]
@jourdain @floryst ?

File: Sources/macro.js [hunk 2]
```diff
@@ -1,6 +1,9 @@
+import globalThisShim from 'globalthis';
 import vtk from './vtk';
 import ClassHierarchy from './Common/Core/ClassHierarchy';
 
+const vtkGlobal = globalThisShim(); // returns native globalThis if compliant
```
Comment (jourdain) [inline Sources/macro.js:5]: can you `export` it so it could be used outside?

File: Sources/vtk.js [hunk 3]
```diff
@@ -1,3 +1,7 @@
+import globalThisShim from 'globalthis';
+
+const vtkGlobal = globalThisShim(); // returns native globalThis if compliant
```
Comment (jourdain) [inline Sources/vtk.js:3]: I would rather use the one from macro once exported. I don't want to spread our dependency to `globalthis` especially if we replace it down the road.

File: Sources/macro.js (hunk 2 above)
Comment (thewtex) [reply Sources/macro.js:5]: Sure

File: Sources/vtk.js (hunk 3 above)
Comment (thewtex) [reply Sources/vtk.js:3]: Ok

File: Sources/macro.js (hunk 2 above)
Comment (thewtex) [reply Sources/macro.js:5]: I had to export from `vtk.js` becase `macro.js` depends on it.

**Comment by jourdain** [comment]
LGTM

**Pull request merged**